source/codegen/proto_helpers.py

```diff
@@ -50,15 +50,16 @@ def is_decomposable_enum(parameter: dict):
   This is because protobuf does not support oneof on repeated types, so the standard
   input decomposition does not work for arrays.
   """
-  return common_helpers.is_enum(parameter) and not is_array_input(parameter)
+  return common_helpers.is_enum(parameter) and not (is_array_input(parameter) and parameter["grpc_type"] != "string")
```

Comment:
There are a handful of changes like this where we were treating `char[]` parameters to functions as arrays. I think we got away with it because most places these are marked as `ViString` and `is_array_input()` returns False for these. Note that I added unit tests for this case.